thunder/tests/test_cudnn_executor.py

```diff
@@ -209,25 +206,25 @@ def test_vjp_correctness_sdpa_cudnnex_manual(op, device, dtype, executor, comp):
             continue
 
         # Compute vjp result using PyTorch
-        expect_out = op.torch_reference(*contiguous_args, **sample.kwargs)
+        expect_out = grad_sdpa_cudnn_opinfo.torch_reference(*contiguous_args, **sample.kwargs)
         v = make_tensor_like(expect_out)
         expected_grad = torch.autograd.grad(expect_out, grad_inputs, v)
 
         # Compute vjp result using Thunder
-        flat_op, flat_args, spec = flatten_func(op.op, sample.args, sample.kwargs)
+        flat_op, flat_args, spec = flatten_func(grad_sdpa_cudnn_opinfo.op, sample.args, sample.kwargs)
         filtered_op, filtered_args = _make_differentiable_wrapper(flat_op, flat_args)
 
         cfoo = thunder.compile(
             vjp(filtered_op),
             disable_torch_autograd_support=True,
             disable_preprocessing=True,
-            executors_list=executor.executors_list() + [cudnn_ex],
+            executors_list=[cudnn_ex],
+            cudnn_sdpa_bwd_may_cat_grad_qkv=may_cat_grad_qkv,
         )
 
         actual_out, actual_grad = cfoo(filtered_args, (v,))
 
-        comp(actual_out, expect_out, atol=1e-2, rtol=1e-2)
-
+        torch.testing.assert_close(actual_out, expect_out, atol=1e-2, rtol=1e-2)
```

Comment:
wondering if tolerance should be part of the opinfo? Not really something that we should do in this PR.

Reply:
It's already allowed to be, e.g., https://github.com/Lightning-AI/lightning-thunder/blob/c67c81428000da9fe26e5c62f750ede9a87ed361/thunder/tests/opinfos.py#L6409. Figuring out the right threshold for cudnn sdpa, however, is real art to me.

Reply:
True.

bf16 tolerances used for internal cudnn testing are also always higher than fp16.
Right now. sdpa tests in cudnn testing do:
`atol=2e-2 if input_type != torch.bfloat16 else 7e-2, rtol=2e-2`

Plus, it is known that there is slightly large mismatches between flash attention based kernels compared to pure math kernels. So depending on what reference is being run above, there are large swings in absolute difference.

Reply:
Got it. Do you request a code change here? I thought I didn't change the behavior but I could be wrong if `comp` (what we used to use) is more complicated than just forwarding to `torch.testing.allclose`.

Reply:
I think its fine as it passes all tests. I can rework tolerances once it starts causing issues.